modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5519,4 +5641,40 @@ public FeatureSet getFeatureSet() {
     public void setFeatureSet(final FeatureSet featureSet) {
         this.featureSet = featureSet == null ? DefaultFeatureSet : featureSet;
     }
+
+    //// Following methods are related to the "useOneOfInterfaces" feature
+    /**
+     * Add "x-oneOf-name" extension to a given oneOf schema (assuming it has at least 1 oneOf elements)
+     * @param s schema to add the extension to
+     * @param name name of the parent oneOf schema
+     */
+    public void addOneOfNameExtension(ComposedSchema s, String name) {
+        if (s.getOneOf() != null && s.getOneOf().size() > 0) {
+            s.addExtension("x-oneOf-name", name);
+        }
+    }
+
+    /**
+     * Add a given ComposedSchema as an interface model to be generated
+     * @param cs ComposedSchema object to create as interface model
+     * @param type name to use for the generated interface model
+     */
+    public void addOneOfInterfaceModel(ComposedSchema cs, String type) {
+        CodegenModel cm = new CodegenModel();
+
+        for (Schema o : cs.getOneOf()) {
+            // TODO: inline objects
```

Comment:
Related to this comment, and line 5672. From https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#discriminatorObject:

> When using the discriminator, inline schemas will not be considered.

I think we'd need to move line 5672 above this loop, and if discriminator is present, don't honor inline models.

Reply:
Good point. I'll take a look and try to address this.

Reply:
I added a commit that handles this. TBH, I found out that this doesn't work with inline models even if the discriminator is not used (because inline models don't have `$ref`). I added a warning about this that would get emitted if inline objects are used.

I honestly don't know ATM how I'd even add the support for inline models to this machinery and don't have too much time to work on that. Hopefully this can be accepted as something to work on going forward and wouldn't be required to accept this PR.

Reply:
looks great!